client/client-multi/client-objective-c/Kaa/DefaultEndpointRegistrationManager.m

```diff
@@ -0,0 +1,292 @@
+                [self.detachEndpointRequests removeObjectForKey:[NSNumber numberWithInt:detached.requestId]];
+            }
+            if (keyHash && detached.result == SYNC_RESPONSE_RESULT_TYPE_SUCCESS) {
+                if ([keyHash isEqual:[self.state endpointKeyHash]]) {
+                    [self.state setIsAttachedToUser:NO];
+                }
+            }
+        }
+    }
+    if (attachNotification) {
+        [self.state setIsAttachedToUser:YES];
+        if (self.attachEndpointToUserDelegate) {
+            __weak typeof(self)weakSelf = self;
+            [[self.context getCallbackExecutor] addOperationWithBlock:^{
+                [weakSelf.attachEndpointToUserDelegate onAttachedToUser:attachNotification.userExternalId
+                                                                  token:attachNotification.endpointAccessToken];
+            }];
+        }
+    }
+    if (detachNotification) {
+        [self.state setIsAttachedToUser:NO];
+        if (self.detachEndpointFromUserDelegate) {
+            __weak typeof(self)weakSelf = self;
+            [[self.context getCallbackExecutor] addOperationWithBlock:^{
+                [weakSelf.detachEndpointFromUserDelegate onDetachedFromUser:detachNotification.endpointAccessToken];
+            }];
+        }
+    }
+}
+
+- (NSDictionary *)getAttachEndpointRequests {
+    @synchronized (self.attachEndpointRequests) {
+        return [NSMutableDictionary dictionaryWithDictionary:self.attachEndpointRequests];
+    }
+}
+
+- (NSDictionary *)getDetachEndpointRequests {
+    @synchronized (self.detachEndpointRequests) {
+        return [NSMutableDictionary dictionaryWithDictionary:self.detachEndpointRequests];
+    }
+}
+
+- (UserAttachRequest *)getUserAttachRequest {
+    return _userAttachRequest;
+}
+
+- (BOOL)isAttachedToUser {
+    return [self.state isAttachedToUser];
+}
+
+- (void)setAttachedDelegate:(id<AttachEndpointToUserDelegate>)delegate {
+    self.attachEndpointToUserDelegate = delegate;
+}
+
+- (void)setDetachedDelegate:(id<DetachEndpointFromUserDelegate>)delegate {
+    self.detachEndpointFromUserDelegate = delegate;
+}
+
+- (void)notifyAttachedDelegate:(SyncResponseResultType)result
+                      delegate:(id<OnAttachEndpointOperationDelegate>)delegate
+                       keyHash:(EndpointKeyHash *)keyHash {
+    if (delegate) {
+        [[self.context getCallbackExecutor] addOperationWithBlock:^{
+            [delegate onAttach:result resultContext:keyHash];
+        }];
+    }
+}
+
+- (void)notifyDetachedDelegate:(SyncResponseResultType)result delegate:(id<OnDetachEndpointOperationDelegate>)delegate {
+    if (delegate) {
+        [[self.context getCallbackExecutor] addOperationWithBlock:^{
+            [delegate onDetach:result];
+        }];
+    }
+}
+
+- (void)onEndpointAccessTokenChanged {
+    if (self.profileTransport) {
+        [self.profileTransport sync];
+    }
+}
+
+- (int)getRandomInteger {
```

Comment:
Amended.
In Java SDK we have the same problem. Maybe we need to open corresponding task on Java SDK?